src/json-api-koa.ts

```diff
@@ -0,0 +1,50 @@
+import { Context } from "koa";
+
+import Application from "./application";
+import { JsonApiDocument, Operation, OperationResponse } from "./types";
+
+export default function jsonApiKoa(app: Application) {
+  const URL_REGEX = new RegExp(app.types.map(t => t.name).join("|"), "i");
+
+  return async (ctx: Context, next: Function) => {
+    if (ctx.url.match(URL_REGEX) && app.types.length) {
+      const ops: Operation[] = convertHttpRequestToOperations(ctx);
+      const result: OperationResponse[] = await app.executeOperations(ops);
+      ctx.body = convertOperationsResponsesToHttpResponse(ctx, result);
+    } else if (ctx.url.match("/bulk")) {
+      return {
+        operations: await app.executeOperations(
+          ctx.request.body.operations || []
+        )
+      };
+    }
+
+    await next();
```

Comment:
@Exelord This should be `return next()`

Reply:
hmm, according to the doc it's ok. Am I missing sth?
```
app.use(async (ctx, next) => {
  const start = Date.now();
  await next();
  const ms = Date.now() - start;
  ctx.set('X-Response-Time', `${ms}ms`);
});
```

Reply:
Probably won't make a difference, nvm :)